src/data/serializer/XYZ.js

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2015 The Open Source Geospatial Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ /**
+  * A serializer for layer that hava a `ol.source.XYZ` source.
+  */
+Ext.define('GeoExt.data.serializer.XYZ', {
+    extend: 'GeoExt.data.serializer.Base',
+    inheritableStatics: {
+        /**
+         * @inheritdoc
+         */
+        sourceCls: ol.source.XYZ,
+
+        /**
+         *
+         */
+        validateSource: function(source) {
+            if (!(source instanceof this.sourceCls)) {
+                Ext.raise("Cannot serialize this source with this serializer");
+            }
```

Comment:
Can the above replaced with:

```javascript
GeoExt.data.serializer.Base.validateSource(source)
```

?

This way the concrete implementation may be changed and we would always get the fresh one.

I am not sure if this works, though.